tests/unit/test_httpchecksum.py

```diff
@@ -301,6 +301,34 @@ def test_apply_request_checksum_flex_header_trailer_explicit_digest(self):
         # The body should not have been wrapped
         self.assertIsInstance(request["body"], bytes)
 
+    def test_apply_request_checksum_content_encoding_preset(self):
```

Comment:
We may want to either do a one-off manual test or add a functional test to ensure operations that bind input to the `Content-Encoding` header aren't affected when a checksum is used. S3 PutObject would be the best place to verify that.

Reply:
Oh sorry I didn't mention in the PR that I did some manual testing making actual `put_object` requests to a test bucket. I did think this may be worthy of a new functional/integration test but wasn't sure.

Checking a stubbed request context/headers should be simple enough to add. I'll update in the morning. I'll also add a changelog unless you don't think it necessary?